mypy/checker.py

```diff
@@ -834,54 +831,47 @@ def check_reverse_op_method(self, defn: FuncItem, typ: CallableType,
                                    [None, None],
                                    AnyType(TypeOfAny.special_form),
                                    self.named_type('builtins.function'))
-        if not is_subtype(typ, method_type):
-            self.msg.invalid_signature(typ, context)
+        if not is_subtype(reverse_type, method_type):
+            self.msg.invalid_signature(reverse_type, context)
             return
 
-        if method in ('__eq__', '__ne__'):
+        if reverse_name in ('__eq__', '__ne__'):
             # These are defined for all objects => can't cause trouble.
             return
 
         # With 'Any' or 'object' return type we are happy, since any possible
         # return value is valid.
-        ret_type = typ.ret_type
+        ret_type = reverse_type.ret_type
         if isinstance(ret_type, AnyType):
             return
         if isinstance(ret_type, Instance):
             if ret_type.type.fullname() == 'builtins.object':
                 return
 
-        if len(typ.arg_types) == 2:
-            # TODO check self argument kind
+        if len(reverse_type.arg_types) != 2:
+            return
```

Comment:
I actually think this should raise an internal error instead of returning, since we should never reach there, now that the signatures are checked.

Reply:
Makes sense to me. Fixed.